### PR TITLE
Fixs #3788: Untranslated strings transfer

### LINKED ISSUE #3788: Break AddProfileActivity string "*Required" into two parts.
`“*Required”` string in `Add Profile` should be broken down into two parts because “*” does not need a translation.

### PR DESCRIPTION
**Pull request by rishidyno**
Fix #3788

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## Explanation

1. Transferred non-translatable strings of `developer_option_activity` to `untranslated_strings.xml` file as they don't need a translation.

2.  Broke `"*Required"` string in `add_profile_activity` into 2 parts .
##

|  LTR Before        |  LTR After |
| -----------       | -----------      |
|cland |cland|

|  RTL Before        |     RTL After       |
| -----------       | -----------      |
|port|port|

### CODE REVIEW

**Comment by rishidyno** [comment]
All tests have passed @rt4914 PTAL

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -326,7 +327,8 @@
   <string name="profile_reset_pin_error_user_pin_length">Your PIN should be 3 digits long.</string>
   <string name="profile_reset_pin_error_admin_pin_length">Your PIN should be 5 digits long.</string>
   <string name="add_profile_create_a_3_digit_pin">Create a 3-Digit PIN</string>
-  <string name="add_profile_required">*Required</string>
+  <string name="add_profile_required">Required</string>
+  <string name="add_profile_required_star">*</string>
```
Comment (rt4914) [inline app/src/main/res/values/strings.xml]: Remove `add_profile_required_star ` as this is not required.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -243,6 +243,7 @@
   <string name="add_profile_error_pin_confirm_wrong">Please make sure that both PINs match.</string>
   <string name="add_profile_info_content_description">More information on 3-digit PINs.</string>
   <string name="required_fields_content_description">Fields marked with an * are required.</string>
+  <string name="required_fields_content_description_star">Asterisk sign</string>
```
Comment (rt4914) [inline app/src/main/res/values/strings.xml]: Remove this, we don't need this.

File: app/src/main/res/layout/add_profile_activity.xml [hunk 3]
```diff
@@ -64,16 +64,33 @@
             app:layout_constraintEnd_toEndOf="@+id/add_profile_activity_user_image_view"
             app:layout_constraintRight_toRightOf="@+id/add_profile_activity_user_image_view" />
 
-          <TextView
-            android:id="@+id/add_profile_activity_required_heading_text_view"
-            android:layout_width="match_parent"
+          <LinearLayout
+            android:id="@+id/add_profile_activity_required_relative_layout"
```
Comment (rt4914) [inline app/src/main/res/layout/add_profile_activity.xml]: Rename to `add_profile_activity_required_text_container`

File: app/src/main/res/layout/add_profile_activity.xml [hunk 4]
```diff
@@ -64,16 +64,33 @@
...
+            <TextView
+              android:id="@+id/add_profile_activity_required_heading_star"
+              android:layout_width="wrap_content"
+              android:layout_height="wrap_content"
+              android:contentDescription="@string/required_fields_content_description_star"
+              android:text="@string/add_profile_required_star"
+              android:textColor="@color/oppiaPrimaryText"
+              android:textSize="12sp" />
+
+            <TextView
```
Comment (rt4914) [inline app/src/main/res/layout/add_profile_activity.xml]: @rishidyno I was just trying out another solution where we can actually solve this without introducing another text view.

Simply in earlier TextView change string from
`android:text="@string/add_profile_required"`
to 
`android:text="@{'*' + @string/add_profile_required}"`

And it should work correctly.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@rishidyno Thanks for the PR. I have suggested changes. Also the screenshots are incorrect as the heading mentions OLD/NEW screenshots but instead they are LTR vs RTL screenshot.


For this PR we just need 4 screenshots:
1. Before - LTR
2. After - LTR
3. Before RTL
4. After RTL

File: app/src/main/res/layout/add_profile_activity.xml (hunk 4 above)
Comment (rishidyno) [reply app/src/main/res/layout/add_profile_activity.xml]: I have made the required changes and it works fine on the device.

File: app/src/main/res/layout/add_profile_activity.xml (hunk 3 above)
Comment (rishidyno) [reply app/src/main/res/layout/add_profile_activity.xml]: Now we only have a text_view its not required

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (rishidyno) [reply app/src/main/res/values/strings.xml]: Done

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (rishidyno) [reply app/src/main/res/values/strings.xml]: Done

**Comment by rishidyno** [comment]
I have made all the changes asked PTAL @rt4914

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

File: app/src/main/res/layout/add_profile_activity.xml [hunk 5]
```diff
@@ -70,9 +70,10 @@
             android:layout_height="wrap_content"
             android:layout_marginTop="32dp"
             android:contentDescription="@string/required_fields_content_description"
-            android:text="@string/add_profile_required"
+            android:text="@{'*' + @string/add_profile_required}"
             android:textColor="@color/oppiaPrimaryText"
             android:textSize="12sp"
+            app:layout_constraintStart_toStartOf="parent"
```
Comment (BenHenning) [inline app/src/main/res/layout/add_profile_activity.xml]: What does this change have to do with moving strings? It seems extra.

File: app/src/main/res/layout/add_profile_activity.xml [hunk 6]
```diff
@@ -70,9 +70,10 @@
             android:layout_height="wrap_content"
             android:layout_marginTop="32dp"
             android:contentDescription="@string/required_fields_content_description"
-            android:text="@string/add_profile_required"
+            android:text="@{'*' + @string/add_profile_required}"
```
Comment (BenHenning) [inline app/src/main/res/layout/add_profile_activity.xml]: This should probably be done in Kotlin, instead, to ensure we are formatting it correctly (we have a lot of bidirectional wrapping stuff coming for proper localization, and this is a case that may be easily missed).

Also, is there a test for this case?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @rishidyno! Just a couple of comments, PTAL.

**Comment by rishidyno** [comment]
I have created a new PR for this issue because some things got mixed up and I had to delete my forked repo and refork. Sorry for the inconvenience.

**Pull request closed without merging**